syz-fuzzer/machine_info.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2020 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"runtime"
+	"strings"
+)
+
+func CollectMachineInfo() ([]byte, error) {
+	if runtime.GOOS != "linux" {
+		return []byte{}, nil
+	}
+
+	type machineInfoFunc struct {
+		name string
+		fn   func() ([]byte, error)
+	}
+
+	allMachineInfo := []machineInfoFunc{
+		{"CPU Info", readCPUInfo},
+		{"KVM", readKVMInfo},
+	}
+
+	var buffer bytes.Buffer
+
+	for _, pair := range allMachineInfo {
+		buffer.WriteString(fmt.Sprintf("[%s]\n", pair.name))
+		data, err := pair.fn()
+		if err != nil {
+			if os.IsNotExist(err) {
+				buffer.WriteString(err.Error() + "\n")
+			} else {
+				return nil, err
+			}
+		} else {
+			buffer.Write(data)
+		}
```

Comment:
It would good for readability to add some separator here.

Reply:
I added some separator so the report now look like this:
```
[CPU INFO]
... content ...
-------------------------
```